test/remote/send_violations_tests.js

```diff
@@ -0,0 +1,212 @@
+process.env.ALLOWED_IPS = ALLOWED_IP
+process.env.MAX_VERIFY_CODES = 2
+process.env.MAX_BAD_LOGINS_PER_IP = 1
+process.env.UID_RATE_LIMIT = 3
+process.env.UID_RATE_LIMIT_INTERVAL_SECONDS = 2
+process.env.UID_RATE_LIMIT_BAN_DURATION_SECONDS = 2
+process.env.RATE_LIMIT_INTERVAL_SECONDS = config.limits.rateLimitIntervalSeconds
+
+// Enable reputation test server
+process.env.REPUTATION_SERVICE_ENABLE = config.reputationService.enable
+process.env.REPUTATION_SERVICE_IP_ADDRESS = config.reputationService.host
+process.env.REPUTATION_SERVICE_PORT = config.reputationService.port
+process.env.REPUTATION_SERVICE_TIMEOUT = config.reputationService.timeout
+
+var testServer = new TestServer(config)
+var reputationServer = new ReputationServerStub(config)
+
+var client = restify.createJsonClient({
+  url: 'http://127.0.0.1:' + config.listen.port
+})
+var reputationClient = restify.createJsonClient({
+  url: 'http://' + config.reputationService.host + ':' + config.reputationService.port
+})
+
+Promise.promisifyAll(client, { multiArgs: true })
+Promise.promisifyAll(reputationClient, { multiArgs: true })
+
+test(
+  'startup',
+  function (t) {
+    testServer.start(function (err) {
+      t.type(testServer.server, 'object', 'test server was started')
+      t.notOk(err, 'no errors were returned')
+      t.end()
+    })
+  }
+)
+
+test(
+  'startup reputation service',
+  function (t) {
+    reputationServer.start(function (err) {
+      t.type(reputationServer.server, 'object', 'test server was started')
+      t.notOk(err, 'no errors were returned')
+      t.end()
+    })
+  }
+)
+
+test(
+  'clear everything',
+  function (t) {
+    mcHelper.clearEverything(
+      function (err) {
+        t.notOk(err, 'no errors were returned')
+        t.end()
+      }
+    )
+  }
+)
+
+test(
+  'sends violation /check resulting in lockout',
+  function (t) {
+    return client.postAsync('/check', { email: TEST_EMAIL, ip: TEST_IP, action: TEST_CHECK_ACTION })
+      .spread(function (req, res, obj) {
+        t.equal(res.statusCode, 200, 'first action noted')
+        t.equal(obj.block, false, 'first action not blocked')
+        return client.postAsync('/check', { email: TEST_EMAIL, ip: TEST_IP, action: TEST_CHECK_ACTION })
+      }).spread(function (req, res, obj) {
+        t.equal(res.statusCode, 200, 'second action noted')
+        t.equal(obj.block, false, 'second action not blocked')
+        return client.postAsync('/check', { email: TEST_EMAIL, ip: TEST_IP, action: TEST_CHECK_ACTION })
+      }).spread(function (req, res, obj) {
+        t.equal(res.statusCode, 200, 'third action attempt noted')
+        t.equal(obj.block, true, 'third action blocked')
+        return Promise.delay(TEST_DELAY_MS)
+      }).then(function () {
+        return reputationClient.getAsync('/mostRecentViolation/' + TEST_IP)
+      }).spread(function (req, res, obj) {
+        t.equal(res.body, '"fxa:request.check.block.recoveryEmailVerifyCode"', 'sends violation when /check')
```

Comment:
nit: Not sure if you want to make this dynamic and build it from `TEST_CHECK_ACTION`.